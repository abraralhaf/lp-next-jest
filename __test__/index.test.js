
import { render, screen } from '@testing-library/react'
import Home from '../pages/index'



describe('Home', () => {

  // it('renders a heading', () => {
  //   render(<Home />)

  //   const heading = screen.getByRole('heading', {
  //     name: /welcome to next\.js!/i,
  //   })

  //   expect(heading).toBeInTheDocument()
  // })


  it('renders homepage unchanged', () => {
    const { container } = render(<Home />)
    expect(container).toMatchSnapshot()
  })

})



// import { test, expect } from '@playwright/test'

// test('should navigate to the about page', async ({ page }) => {
//   // Start from the index page (the baseURL is set via the webServer in the playwright.config.ts)
//   await page.goto('http://localhost:3000/')
//   // Find an element with the text 'About Page' and click on it
//   await page.click('text=Mulai')
//   // The new url should be "/about" (baseURL is used there)
//   await expect(page).toHaveURL('#')
//   // The new page should contain an h1 with "About Page"
//   // await expect(page.locator('h1')).toContainText('About Page')
// })

// import React from "react";
// import ReactDOM from "react-dom";
// import Index from '../pages/index.js';
// import { render, screen } from '@testing-library/react';

// test('renders learn react link', () => {
//     render(<Index />);
//     const linkElement = screen.getByText(/learn react/i);
//     expect(linkElement).toBeInTheDocument();
//   });

  
// test("renders the conntent", () =>{
   
//     const root =document.createElement("div");
//     ReactDOM.render(<Index />, root);
// });

// describe('Renders the Content', () => {
//     test('load', ()=>{

//         render(<Index />);
//         const linkElement = screen.getByText(/learn react/i);
//         expect(linkElement).toBeInTheDocument();

//         // const wrapper = shallow(<Benefit></Benefit>);
//         // const text = wrapper.find('p p');
//         // expect(text.text()).toBe('Dengarkan dimana saja')
//     }); 
    
// })































