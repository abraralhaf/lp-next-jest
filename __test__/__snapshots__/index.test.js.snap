// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <div
    class="main"
  >
    <nav
      class="nav bg-slate-900 sticky top-0 z-50"
    >
      <div
        class="main-container gap-x-6 justify-between h-full w-full "
      >
        <div>
          <a
            href="/"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27170%27%20height=%2730%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Spotify Inc."
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </a>
        </div>
        <div>
          <ul
            class="flex items-center gap-x-7"
          >
            <li
              class="hidden sm:list-item"
            >
              <a
                class="text-header text-sm hover:text-green-500"
                href="/#"
              >
                Premium
              </a>
            </li>
            <li
              class="hidden sm:list-item"
            >
              <a
                class="text-header text-sm hover:text-green-500"
                href="/#"
              >
                Dukungan
              </a>
            </li>
            <li
              class="hidden sm:list-item"
            >
              <a
                class="text-header text-sm hover:text-green-500"
                href="/#"
              >
                Download
              </a>
            </li>
            <li>
              <button
                class="text-header button-login text-sm hover:text-green-500"
                href="/#"
              >
                Masuk
              </button>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <header
      class="header"
    >
      <div
        class="main-container p-20 mt-8 header-bg"
      >
        <div
          class="w-3/6 ml-10 md:w-3/6"
        >
          <h1
            class="text-header text-4xl mb-4"
          >
            Dapatkan 2 bulan Premium cuma Rp 54,990 dengan e-wallet dan pulsa.
          </h1>
          <h2
            class="text-white/80 text-md"
          >
            Hanya Rp 54,990/bulan sesudahnya. Batalkan kapan saja.
          </h2>
          <div
            class="hidden sm:flex items-center gap-x-2 py-4"
          >
            <button
              class="font-bold text-black text-sm button-rounded border border-green-600/75 bg-green-600 px-6 py-1 rounded-full shadow-sm hover:shadow-xl px-7 py-2"
            >
              MULAI
            </button>
            <button
              class="font-semi-bold text-white/80 text-sm button-rounded px-6 py-1 rounded-full shadow-sm border border-white/80 hover:shadow-xl px-7 py-2"
            >
              LIHAT PAKET
            </button>
          </div>
          <p
            class="text-white/80 text-xs"
          >
            <u>
              Persyaratan dan ketentuan berlaku.
            </u>
             Tawaran tidak tersedia untuk pengguna yang pernah mencoba Premium.
          </p>
        </div>
      </div>
      <div
        class="main-container p-16"
      >
        <div
          class="w-full"
        >
          <h1
            class="text-header text-center text-3xl mb-2"
          >
            Memperkenalkan Premium Mini
          </h1>
          <h2
            class="text-white text-center"
          >
            Download 30 lagu dan dengarkan musik bebas iklan semaumu, di 1 ponsel
          </h2>
          <div
            class="flex justify-center gap-x-5 py-4"
          >
            <button
              class="font-bold text-black text-sm button-rounded border border-gray-300 bg-gray-300 px-6 py-1 rounded-full shadow-sm px-7 py-2 hover:shadow-xl"
            >
              1 HARI RP 2,500
            </button>
            <button
              class="font-semi-bold text-white/80 text-sm button-rounded px-6 py-1 rounded-full shadow-sm border border-white/80 hover:shadow-xl px-7 py-2"
            >
              1 MINGGU RP 9,500
            </button>
          </div>
          <p
            class="text-white/80 text-xs text-center"
          >
            <u>
              Persyaratan dan ketentuan berlaku.
            </u>
          </p>
        </div>
      </div>
      <div
        class="border-t border-slate-700"
      />
    </header>
    <section>
      <div
        class="main-container py-10"
      >
        <div
          class="w-full"
        >
          <h1
            class="text-header text-center text-3xl"
          >
            Kenapa beli Premium?
          </h1>
          <div
            class="flex justify-center grid grid-cols-1 gap-4 mt-10 ml-10 md:grid-cols-2 lg:grid-cols-4 "
          >
            <div
              class="bg-slate-800 p-5 rounded-lg"
            >
              <div
                class="text-center py-8"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2754%27%20height=%2754%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Download"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <h1
                class="text-white font-semibold text-center"
              >
                Download Musik
              </h1>
              <p
                class="text-white text-center text-sm"
              >
                Dengarkan dimana saja
              </p>
            </div>
            <div>
              <div
                class="bg-slate-800 p-5 rounded-lg"
              >
                <div
                  class="text-center py-8"
                >
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2754%27%20height=%2754%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="Download"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </div>
                <h1
                  class="text-white font-semibold text-center"
                >
                  Dengarkan musik bebas iklan.
                </h1>
                <p
                  class="text-white text-center text-sm"
                >
                  Nikmati musik tanpa henti
                </p>
              </div>
            </div>
            <div>
              <div
                class="bg-slate-800 p-5 rounded-lg"
              >
                <div
                  class="text-center py-8"
                >
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2754%27%20height=%2754%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="Star"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </div>
                <h1
                  class="text-white font-semibold text-center"
                >
                  Putar lagu apa pun
                </h1>
                <p
                  class="text-white text-center text-sm"
                >
                  Bahkan di HP.
                </p>
              </div>
            </div>
            <div>
              <div
                class="bg-slate-800 p-5 rounded-lg"
              >
                <div
                  class="text-center py-8"
                >
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2754%27%20height=%2754%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="Play"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </div>
                <h1
                  class="text-white font-semibold text-center"
                >
                  Lewati sesukamu
                </h1>
                <p
                  class="text-white text-center text-sm"
                >
                  Cukup tekan berikutnya.
                </p>
              </div>
            </div>
          </div>
          <div
            class="p-5 mt-5"
          >
            <div
              class="flex justify-center items-center gap-x-5 py-4"
            >
              <h2
                class="text-white font-semibold text-xl"
              >
                Diskon khusus untuk mahasiswa yang memenuhi syarat
              </h2>
              <button
                class="text-white button-rounded text-sm border rounded-full px-7 py-2"
              >
                Pelajari
                <br />
                 Selengkapnya
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="border-t border-slate-700"
      />
    </section>
    <footer
      class="bg-black mt-10"
    >
      <div
        class="main-container py-20"
      >
        <div
          class="grid grid-cols-1 ml-10 gap-4 md:grid-cols-3 lg:grid-cols-5"
        >
          <a
            href="/#"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27170%27%20height=%2730%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Spotify"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </a>
          <div>
            <ul
              class="gap-2"
            >
              <li>
                <p
                  class="uppercase font-semibold text-white/80 text-[12px] mb-2"
                >
                  Perusahaan
                </p>
              </li>
              <li>
                <a
                  class="text-footer-item"
                  href="/#"
                >
                  Tentang
                </a>
              </li>
              <li>
                <a
                  class="text-footer-item"
                  href="/#"
                >
                  Pekerjaan
                </a>
              </li>
              <li>
                <a
                  class="text-footer-item"
                  href="/#"
                >
                  For the Record
                </a>
              </li>
            </ul>
          </div>
          <div>
            <ul
              class="gap-2"
            >
              <li>
                <p
                  class="uppercase font-semibold text-white/80 text-[12px] mb-2"
                >
                  Komunitas
                </p>
              </li>
              <li>
                <a
                  class="text-footer-item"
                  href="/#"
                >
                  Untuk Artis
                </a>
              </li>
              <li>
                <a
                  class="text-footer-item"
                  href="/#"
                >
                  Pengembang
                </a>
              </li>
              <li>
                <a
                  class="text-footer-item"
                  href="/#"
                >
                  Iklan
                </a>
              </li>
              <li>
                <a
                  class="text-footer-item"
                  href="/#"
                >
                  Investor
                </a>
              </li>
              <li>
                <a
                  class="text-footer-item"
                  href="/#"
                >
                  Vendor
                </a>
              </li>
            </ul>
          </div>
          <div>
            <ul
              class="gap-2"
            >
              <li>
                <p
                  class="uppercase font-semibold text-white/80 text-[12px] mb-2"
                >
                  Tautan Berguna
                </p>
              </li>
              <li>
                <a
                  class="text-footer-item"
                  href="/#"
                >
                  Dukungan
                </a>
              </li>
              <li>
                <a
                  class="text-footer-item"
                  href="/#"
                >
                  Pemutar Web
                </a>
              </li>
              <li>
                <a
                  class="text-footer-item"
                  href="/#"
                >
                  Aplikasi Seluler Gratis
                </a>
              </li>
            </ul>
          </div>
          <div>
            <ul
              class="flex justify-end gap-x-16 gap-2"
            >
              <li>
                <a
                  class="text-footer-item"
                  href="/#"
                >
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2732%27%20height=%2732%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="Instagram"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </a>
              </li>
              <li>
                <a
                  class="text-footer-item"
                  href="/#"
                >
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2732%27%20height=%2732%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="Twitter"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </a>
              </li>
              <li>
                <a
                  class="text-footer-item"
                  href="/#"
                >
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2732%27%20height=%2732%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="Facebook"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="main-container pb-10 ml-10"
      >
        <div
          class="flex items-center gap-x-6 ml-10"
        >
          <a
            class="text-footer-item-2"
            href="/#"
          >
            Hukum
          </a>
          <a
            class="text-footer-item-2"
            href="/#"
          >
            Pusat Privasi
          </a>
          <a
            class="text-footer-item-2"
            href="/#"
          >
            Kebijakan Privasi
          </a>
          <a
            class="text-footer-item-2"
            href="/#"
          >
            Cookie
          </a>
          <a
            class="text-footer-item-2"
            href="/#"
          >
            Tentang Iklan
          </a>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
